commands/Moderation/role.js

```diff
@@ -0,0 +1,29 @@
+exports.run = async (client, msg, [user, role]) => {
+  if (msg.guild.roles.exists("name", role)) {
```

Comment:
Collection#exists followed by Collection#find kills the performance. Find it and check its existence. Also, you might want to make it case insensitive.

Reply:
Use Komada's Role usage instead, then all you need to do is check `if (role)`

Reply:
I think both are needed no?
First, it checks if the role exists on the server. 
Then it checks if the user has the role: A) if yes -> remove the role B) if it doesn't have added role
third, if the role doesn't exist on the server send a message saying role doesn't exist

If i merge it into 1 if statement to check server and role ill have to make multiple if statements
1) Check if role exists && user has role
2) check if role exists && user doesnt have role
3) if role doesnt exist send error message

Reply:
I mean, you could just use Komada's internal parser for the role type, which resolves mention, id, and iirc, also names. 

Map.prototype.has is dozen times faster than `Collection.prototype.exists` and therefore `Collection.prototype.find`.

You get the role object, then check if the user **has** the role, if so, give the role. 

What you're doing is checking a roles collection and get a Boolean from there (`Collection.prototype.exists`), then check if the user has said role with the same name, this is unaccurate as there can be two roles with the same name, and in Guild.roles it'd get one role but in GuildMember.roles get another. 

Then get the same value you have checked, for second time, to apply the role to the user. 

You're doing three checks, twice for the same thing in the same collection. 

The preferable step is to make Komada get the role internally throught the parser, you know 100% it exists, and Komada can use cmdPrompt in case you entered a wrong role name, as opposed to writting it again (or editing it if cmdEditing is set to true), check if the GuildMember has said role by checking the ID, which is extremely fast, and add the role straight-up from there, without fetching it again from the collection.